library/core/src/array/mod.rs

```diff
@@ -364,3 +364,61 @@ macro_rules! array_impl_default {
 }
 
 array_impl_default! {32, T T T T T T T T T T T T T T T T T T T T T T T T T T T T T T T T}
+
+#[cfg(not(bootstrap))]
+#[lang = "array"]
+impl<T, const N: usize> [T; N] {
+    /// Returns an array of the same size as `self`, with `f` applied to each element.
+    /// The closure will be called on elements 0 up to N.
```

Comment:
I think we should say excluding `N`, "up to" may not be that clear and may be interpreted as including.